x-pack/plugins/uptime/public/state/effects/journey.ts

```diff
@@ -16,16 +16,23 @@ import {
 import { fetchJourneySteps } from '../api/journey';
 import type { SyntheticsJourneyApiResponse } from '../../../common/runtime_types';
 
+const inFlightStepRequests: Record<FetchJourneyStepsParams['checkGroup'], boolean> = {};
+
 export function* fetchJourneyStepsEffect(): Generator {
-  yield takeLatest(getJourneySteps, function* (action: Action<FetchJourneyStepsParams>) {
+  yield takeEvery(getJourneySteps, function* (action: Action<FetchJourneyStepsParams>) {
```

Comment:
You can get rid of all the extra logic and replace the original `takeLatest` with `takeLeading`. https://redux-saga.js.org/docs/api/#takeleadingpattern-saga-args

Reply:
Thank you for the suggestion, I think what we're doing here is a bit different than just picking the leading request though.

I have discussed this with Shahzad before, and the problem with `takeLeading` is that it will take the leading dispatch action no matter what ID you're fetching. That means that if you fetch checkgroup A and, while fetching checkgroup A, try to fetch checkgroup B, you will _not_ fetch checkgroup B.

The reason we have this logic is so that you only have _one inflight request **per ID**_.

Imagine the following problematic scenario, for example:
1. Page 1 starts fetching checkgroup A
2. User navigates to Page 2 which tries to start fetching checkgroup B
3. takeLeading blocks the checkgroup B and just continues fetching checkgroup A instead
4. checkgroup B doesn't load until A is complete and user navigates back